src/views/visual/visual-view-graph.js

```diff
@@ -54,41 +116,34 @@ export class VisualViewGraph extends LitElement {
       selected: this.selectedCollections,
       language: this.language,
     });
-
-    // These chartHeight values are just guesses. It might have to be adjusted when more data is loaded.
-    // When a good way of dealing with this is found, this function needs clearning up too.
-    this.chartHeight = '';
-    let heightMultiplyFactor = graphdata.length / window.innerHeight;
-    if (heightMultiplyFactor <= 0.5) {
-      this.chartHeight = '84vh';
-    } else if (heightMultiplyFactor <= 1) {
-      this.chartHeight = graphdata.length * heightMultiplyFactor * 3 + 'px';
-    } else if (heightMultiplyFactor <= 1.5) {
-      this.chartHeight = graphdata.length * heightMultiplyFactor * 2 + 'px';
-    } else if (heightMultiplyFactor <= 2) {
-      this.chartHeight =
-        (graphdata.length * heightMultiplyFactor * 3) / 2 + 'px';
-    } else if (heightMultiplyFactor <= 3) {
-      this.chartHeight = graphdata.length * heightMultiplyFactor + 'px';
-    } else if (heightMultiplyFactor <= 4) {
-      this.chartHeight = graphdata.length / 2 + 'px';
-    } else if (heightMultiplyFactor <= 5) {
-      this.chartHeight = graphdata.length / 2.5 + 'px';
-    } else {
-      this.chartHeight = '20000px';
-    }
-    this.graphData = graphdata;
+    this.graphData = this.paginateGraphData(graphdata);
+    this.totalPages = this.graphData.length;
+    this.adjustChartHeight();
     this.fetchError = error;
     this.fetchLoading = false;
   }
 
+  adjustChartHeight() {
+    this.chartHeight = '84vh';
+    if (
+      this.graphData &&
+      this.graphData[this.currentPage] &&
+      this.graphData[this.currentPage].length * 2 > 800
+    ) {
+      this.chartHeight = this.graphData[this.currentPage].length * 2 + 'px';
```

Comment:
this is better, but can it not be solved with CSS? What's the issue we're trying to solve here?

Reply:
Not exactly. It's a difficult problem where we are constantly experimenting with the height of the chart depending on the size of the dataset.